### PR TITLE
chore: YARA 4.1.0 compliance

### PR DESCRIPTION
**Pull request by secDre4mer**
Adjust code and tests to return a file_size for custom memory
iterators.

### CODE REVIEW

File: mem_blocks.go [hunk 1]
```diff
@@ -29,6 +30,7 @@ import (
 type MemoryBlockIterator interface {
 	First() *MemoryBlock
 	Next() *MemoryBlock
+	Filesize() uint64
```
Comment (hillu) [inline mem_blocks.go:33]: Iterators are not required to implement a file size function (`iterator->file_size` may be `NULL`), so this should go into a separate optional interface.

File: mem_blocks.go [hunk 2]
```diff
@@ -60,9 +62,10 @@ func makeMemoryBlockIteratorContainer(mbi MemoryBlockIterator) (c *memoryBlockIt
 // pool.
 func makeCMemoryBlockIterator(c *memoryBlockIteratorContainer) (cmbi *C.YR_MEMORY_BLOCK_ITERATOR) {
 	return &C.YR_MEMORY_BLOCK_ITERATOR{
-		context: callbackData.Put(c),
-		first:   C.YR_MEMORY_BLOCK_ITERATOR_FUNC(C.memoryBlockIteratorFirst),
-		next:    C.YR_MEMORY_BLOCK_ITERATOR_FUNC(C.memoryBlockIteratorNext),
+		context:   callbackData.Put(c),
+		first:     C.YR_MEMORY_BLOCK_ITERATOR_FUNC(C.memoryBlockIteratorFirst),
+		next:      C.YR_MEMORY_BLOCK_ITERATOR_FUNC(C.memoryBlockIteratorNext),
+		file_size: C.YR_MEMORY_BLOCK_ITERATOR_SIZE_FUNC(C.memoryBlockIteratorFilesize),
```
Comment (hillu) [inline mem_blocks.go:68]: …and `file_size` should only be set if a FileSize implementation exists.

**Review by hillu — CHANGES_REQUESTED** [review]
Thanks for your PR!

I'd really like to keep API compatibility with go-yara 4.0.x and just take the Semver bump to 4.1, not to 5.0.

The YARA version check in `cgo.go` also needs to be updated, too, because the referencing `iterator->file_size` causes builds to fail with YARA 4.0.x.

File: mem_blocks.go (hunk 1 above)
Comment (secDre4mer) [reply mem_blocks.go:37]: Done

File: mem_blocks.go (hunk 2 above)
Comment (secDre4mer) [reply mem_blocks.go]: Done

**Comment by hillu** [comment]
I have made a minor change to the version check and put pulled everything  together into a single commit. Thank you very much.

**Pull request closed without merging**